cpp/src/arrow/dataset/partition.h

```diff
@@ -321,6 +330,30 @@ class ARROW_DS_EXPORT FunctionPartitioning : public Partitioning {
   std::string name_;
 };
 
+class ARROW_DS_EXPORT FilenamePartitioning : public KeyValuePartitioning {
+ public:
+  /// If a field in schema is of dictionary type, the corresponding element of
+  /// dictionaries must be contain the dictionary of values for that field.
```

Comment:
```suggestion
  /// \brief Construct a FilenamePartitioning from its components.
  ///
  /// If a field in the `schema` is of dictionary type, the corresponding element of
  /// `dictionaries` must contain the dictionary of values for that field.
```

Reply:
Made the change.

Reply:
It seems you didn't commit the change, though?

Reply:
Yes, sorry, missed that. Mistakenly, did the change in the file present in the build include dir. Committed a change now. Thanks for pointing that out!